0005-ordered-quantification.md

```diff
@@ -386,28 +403,28 @@ set comprehensions are encoded as their indicator predicates, but there is no su
 for sequence comprehensions as defined here. The initial encoding will likely be with a function,
 similar to the `Seq#Create` function used for seq constructions, that accepts a source sequence and indicator predicate.
 The useful axioms about these values will fall into three categories, sketched below for the simple
-case of comprehensions with a single variable bound from a sequence and no explicit term:
+case of comprehensions with a single variable bound from a sequence:
 
-`var S := seq x <- C | P(x);`
+`var S := seq x <- C | P(x) :: Q(x);`
 
-1. Membership: `forall x :: P(x) <==> x in S`
+1. Membership: `forall x :: x in C && P(x) ==> Q(x) in S`
 2. Cardinality: `|S| <= |C|`
 3. Ordering: `forall i, j | 0 <= i < j < |S| :: 
       exists i', j' | 0 <= i' < j' < |C| :: 
-        C[i'] == S[i] && C[j'] == S[j]`
+        Q(C[i']) == S[i] && Q(C[j']) == S[j]`
```

Comment:
Could you have a new construct that says:
```
countIf(x <- C, P(x))
```
that is always between |0| and |C|, that increases with C,
that counts the number of P(x) that hold for all the `x` in `C` ?
That way, you wouldn't need an existential inside the forall:

```
Cardinality: `|S| == countIf(x <- C, Q(x))`
Ordering: 
  forall i <
  Q(C[i']) == S[i]
forall i | 0 <= i < |S| :: 
  var i' := countIf(x <- C[0..i], P(x));
  Q(C[i']) == S[i]
```
Your axioms could be deduced from that.